lib/ts/recipe/session/jwt.ts

```diff
@@ -45,12 +45,21 @@ export function parseJWTWithoutSignatureVerification(jwt: string): ParsedJWTInfo
         throw new Error("Invalid JWT");
     }
 
+    // V1&V2 is functionally identical, plus all legacy tokens should be V2 now.
+    let version = 2;
+    // V2 or older tokens did not save the key id;
     // checking header
     if (!HEADERS.has(splittedInput[0])) {
-        throw new Error("JWT header mismatch");
+        const parsedHeader = JSON.parse(Buffer.from(splittedInput[0], "base64").toString());
+        if (parsedHeader.typ !== "JWT" || parsedHeader.version !== "3" || parsedHeader.kid === undefined) {
```

Comment:
!== "3" -> should be < 3?

Reply:
I think it's better to be checking the string match here:
- If we add support for future access token versions, we likely should add new validation rules anyway
- it's a string that we do not need to parse here.

Reply:
There could be future versions added that is still a JWT but have some other claim (similar to refreshTokenHash1) which is required. So this file doesn't change, but instead the accessToken file will only change

Reply:
> it's a string that we do not need to parse here.

You already parse it: `version = Number.parseInt(parsedHeader.version)`